src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -148,21 +152,28 @@ def is_key_or_prefix_key_in_dict(key: Tuple[str]) -> bool:
 def convert_pytorch_state_dict_to_flax(pt_state_dict, flax_model):
     # convert pytorch tensor to numpy
     # numpy currently does not support bfloat16, need to go over float32 in this case to not lose precision
-    try:
-        import torch  # noqa: F401
-    except (ImportError, ModuleNotFoundError):
-        logger.error(
-            "Loading a PyTorch model in Flax, requires both PyTorch and Flax to be installed. Please see"
-            " https://pytorch.org/ and https://flax.readthedocs.io/en/latest/installation.html for installation"
-            " instructions."
-        )
-        raise
 
-    weight_dtypes = {k: v.dtype for k, v in pt_state_dict.items()}
-    pt_state_dict = {
-        k: v.numpy() if not v.dtype == torch.bfloat16 else v.float().numpy() for k, v in pt_state_dict.items()
-    }
+    def is_bfloat16(value):
+        from_bin = is_torch_available() and isinstance(value, torch.Tensor)
+        if from_bin:
+            bfloat16 = torch.bfloat16
+        else:
+            bfloat16 = "bfloat16"
+
+        return value.dtype == bfloat16
+
+    def numpify_state_dict(_pt_state_dict):
+        from_bin = is_torch_available() and isinstance(next(iter(_pt_state_dict.values())), torch.Tensor)
+        _state_dict = {}
+        for k, v in _pt_state_dict.items():
+            if is_bfloat16(v):
+                _state_dict[k] = v.float().numpy() if from_bin else v
+            else:
+                _state_dict[k] = v.numpy() if from_bin else v
+
+        return _state_dict
```

Comment:
Suggestion to remove the two shallow functions and perform everything in the main body of the function. To me this is clearer than the two new function definitions!
```suggestion
    from_bin = is_torch_available() and isinstance(next(iter(pt_state_dict.values())), torch.Tensor)
    bfloat16 = torch.bfloat16 if from_bin else "bfloat16"

    for k, v in pt_state_dict.items():
        if v.dtype == bfloat16:
            pt_state_dict[k] = v.float().numpy() if from_bin else v
        else:
            pt_state_dict[k] = v.numpy() if from_bin else v
```